src/univariate/continuous/lognormal.jl

```diff
@@ -41,6 +52,13 @@ LogNormal(μ::Real, σ::Real; check_args::Bool=true) = LogNormal(promote(μ, σ)
 LogNormal(μ::Integer, σ::Integer; check_args::Bool=true) = LogNormal(float(μ), float(σ); check_args=check_args)
 LogNormal(μ::Real=0.0) = LogNormal(μ, one(μ); check_args=false)
 
+# for arbitrary base
+function LogNormal(μ::Real, σ::Real, B::Real; check_args::Bool=true)
+    @check_args LogNormal (B, B ≥ zero(B))
+    lb = log(B)
+    LogNormal(promote(μ * lb, σ * lb)...; check_args=check_args)
+end
```

Comment:
This could be simplified to
```suggestion
function LogNormal(μ::Real, σ::Real, B::Real; check_args::Bool=true)
    @check_args LogNormal (B, B ≥ zero(B))
    lb = log(B)
    LogNormal(μ * lb, σ * lb; check_args=check_args)
end
```
as the `LogNormal` constructors already take care of promotion.

However, I wonder if we should promote all arguments first to ensure that e.g. `LogNormal(0f0, 1f0, 10)` returns a `LogNormal{Float32}` (currently it would return a `LogNormal{Float64}`). I.e., maybe we should use
```suggestion
function LogNormal(μ::Real, σ::Real, B::Real; check_args::Bool=true)
    return LogNormal(promote(μ, σ, B)...; check_args=check_args)
end
function LogNormal(μ::T, σ::T, B::T; check_args::Bool=true) where {T<:Real}
    @check_args LogNormal (B, B ≥ zero(B))
    lb = log(B)
    LogNormal(μ * lb, σ * lb; check_args=check_args)
end
```

Reply:
Thanks for the review. Good catch; I prefer the latter solution. Will change this and add a test for this behavior.